### DISCUSSION TITLE
Saving network image override the old one!

### DISCUSSION THREAD

**Discussion by mNouh94** [natsuk4ze/gal#186] [Q&A] [answered]
When i save the image from network it overrides the old image and the name changes in the gallery but the old one gets deleted from gallery only it appears in the cache, even I tried to make a different names for each image it does the same!

`final String randomNum = Random().nextInt(100).toString();
      final String imageName = imageUrl.split('/').last;
      final String fileName = '${randomNum}_$imageName';

      final imagePath = '${Directory.systemTemp.path}/$fileName';

      debugPrint('imagePath ==> $imagePath');

      await _apiService.download(
        downloadUrl: imageUrl,
        savePath: imagePath,
      );

      try {
        await Gal.putImage(
          imagePath,
          album: AppConstants.appNameEn,
        );
      } on GalException catch (e) {
        log('error ==> $e');
      }`

**Comment by natsuk4ze** [comment, marked as answer]
Hi @mNouh94 🤟

Is that a Google Photos thing on Android and doesn't happen on iOS?

It probably is.

### Google Photos seems to display images with the same content on the same card. In other words, old images are not deleted. That can be checked from the Android file manager.

It is not a gal issue, as the same behavior can be confirmed by downloading the same image from the internet using chrome on an Android device.

I don't know if there is a setting to stop this stacking of images, but there might be.

💡 Incidentally, gal has the ability to automatically generate a different file name for the same file name to avoid collisions, so there is no need to set the name just to make it unique.

### Finally, don't forget to "Mark as answer ✅". Thank you.

**Reply by mNouh94** [reply]
Hi @natsuk4ze 

The one i have tried before was an emulator.

i tried it from a different real device it works fine, about iOS I have not tried it for now.

Thanks for your answer!
